forem/forem
# Save Confirmation - Admin (Part 1 of 2)

## What type of PR is this? (check all applicable)


## Description
Part 1 of 2 implementing the [Admin-Save-Confirmation RFC](https://github.com/forem/rfcs/blob/237480c4b915a64fc17df8f525b0a429185a1605/text/0078-admin-save-confirmation.md).
In this PR, I refactor all confirmation steps in the `/admin/customization/config` page to use async calls. I also update all associated `requests` and end-to-end tests.

### NOTES
- Do not let the number of `Files changed` alarm you 😅  For about 10 files, I merely added the `Update Settings` button, and added the `updateConfigurationSettings` callback to the form therein.
- It has been observed that there is no visual indication if an update failed or didn't "take". I have noted this and will address this UX-improvement in a future PR. 👍🏾 

---


## QA Instructions, Screenshots, Recordings

1. Login to `http://localhost:xxxx/admin/customization/config` as a `super_admin`
2. Make various changes to the settings on this page. After clicking the `Update Settings` button each time, there should be NO page refresh, and a snackbar with message `Successfully updated settings` should appear in the bottom-left corner.
3. Run `spec/requests/admin/configs_spec.rb` and all tests under `cypress/integration/seededFlows/adminFlows/config` and make sure they pass.

### UI accessibility concerns?
None. No major UI changes made; most changes are backend

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will update the Admin Guide
      replace this line with details on why this change doesn't need to be
      shared_

## What gif best describes this PR or how it makes you feel?

![Captain Marvel](https://media.giphy.com/media/ZvkFmclQO1ImmRNm0K/giphy-downsized.gif?cid=ecf05e477ld684dzxlok1zvg25q9qkpncuffcqkzf6ygdvrp&rid=giphy-downsized.gif&ct=g)
